### PR TITLE
UNOMI-449 : forward existing endpoints to jax-rs endpoints

### PR DESCRIPTION
**Pull request by dgriffon**
see https://issues.apache.org/jira/browse/UNOMI-449 for details

### CODE REVIEW

File: rest/src/main/java/org/apache/unomi/rest/EventsCollectorEndpoint.java [hunk 1]
```diff
@@ -154,11 +139,13 @@ private EventCollectorResponse doEvent(String eventsCollectorRequestAsString, Lo
             */
         } else {
             Profile sessionProfile = session.getProfile();
+            final String errorMessage = String.format("No valid profile found or persona found for profileId=%s, aborting request !", session.getProfileId());
             if (sessionProfile.getItemId() != null) {
                 // Reload up-to-date profile
                 profile = profileService.load(sessionProfile.getItemId());
                 if (profile == null || profile instanceof Persona) {
-                    throw new InternalServerErrorException(String.format("No valid profile found or persona found for profileId=%s, aborting request !", session.getProfileId()));
+                    logger.error(errorMessage);
```
Comment (Taybou) [inline rest/src/main/java/org/apache/unomi/rest/EventsCollectorEndpoint.java:147]: maybe protect this message with "if (logger.isDebugEnabled()) {"

File: wab/src/main/java/org/apache/unomi/web/ContextServlet.java [hunk 2]
```diff
@@ -71,396 +53,21 @@ public void init(ServletConfig config) throws ServletException {
...
-                if (contextRequest.getProfileOverrides().getProperties()!=null) {
-                    profile.setProperties(contextRequest.getProfileOverrides().getProperties());
-                }
-                if (contextRequest.getSessionPropertiesOverrides()!=null && session != null) {
-                    session.setProperties(contextRequest.getSessionPropertiesOverrides());
-                }
-            }
+            response.flushBuffer();
+            logger.debug("OPTIONS request received. No context will be returned.");
+            return;
```
Comment (jkevan) [inline wab/src/main/java/org/apache/unomi/web/ContextServlet.java:61]: Options should also be implemented in the REST endpoint, so like the event collector, we could just forward the request and implement the option in the endpoint

File: wab/src/main/java/org/apache/unomi/web/EventsCollectorServlet.java [hunk 3]
```diff
@@ -61,182 +44,23 @@ public void destroy() {
...
-        } catch (Throwable t) { // Here in order to return generic message instead of the whole stack trace in case of not caught exception
-            logger.error("EventsCollectorServlet failed to execute post", t);
-            resp.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Internal server error");
-        }
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        HttpServletRequestForwardWrapper.forward(request, response);
     }
 
     @Override
     protected void doOptions(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
```
Comment (jkevan) [inline wab/src/main/java/org/apache/unomi/web/EventsCollectorServlet.java:57]: Why not using the options implem of the endpoint by just forwarding the request here ?

**Review by jkevan — APPROVED** [review]

File: rest/src/main/java/org/apache/unomi/rest/EventsCollectorEndpoint.java (hunk 1 above)
Comment (dgriffon) [reply rest/src/main/java/org/apache/unomi/rest/EventsCollectorEndpoint.java:147]: This one will be protected higher in the chain, meaning we can keep it as it is

File: rest/src/main/java/org/apache/unomi/rest/ContextJsonEndpoint.java [hunk 4]
```diff
@@ -82,9 +78,22 @@
...
+    @OPTIONS
+    @Path("/context.json")
+    public Response options() {
+        return Response.status(Response.Status.NO_CONTENT).header("Access-Control-Allow-Origin", "*").build();
+    }
+
     @POST
     @Path("/context.json")
-    public ContextResponse getContextJSON(String contextRequestAsString, @QueryParam("timestamp") Long timestampAsLong, @CookieParam("context-profile-id") String cookieProfileId) {
+    public ContextResponse getContextJSON(
```
[the remark this answers is not in the document]
Comment (dgriffon) [reply rest/src/main/java/org/apache/unomi/rest/ContextJsonEndpoint.java]: done

**Review by jkevan — APPROVED** [review]

**Pull request merged**